cibuildwheel/oci_container.py

```diff
@@ -133,14 +183,30 @@ def __enter__(self) -> Self:
         if detect_ci_provider() == CIProvider.travis_ci and platform.machine() == "ppc64le":
             network_args = ["--network=host"]
 
+        platform_args = [f"--platform={self.oci_platform.value}"]
+        if not self.image.endswith(":cibw_local"):
```

Comment:
rather than forcing the tag to be `cibw_local`, we could use a suffix or prefix that gets removed.
e.g. `cibw_local+myimage:latest` or `myimage:latest+ cibw_local`